Hash/Lessons42579/chan1_solution.py

```diff
@@ -0,0 +1,34 @@
+def solution(genres, plays):
+    answer = []
+    a={}
+    for i in range(len(genres)):
+        
+    #해시 형태 classic : {[500,0], [150,2] ,[800,3]}   
+        
+        temp={}
+        temp[i]=plays[i]
+        if genres[i] in a.keys():
+            a[genres[i]].update(temp)   
+        else:
+            a[genres[i]]=temp
+            
+    #print(a)  
+    #print(a.keys())  
+    
+    
+    result_list=sorted(a.keys(),key=lambda x : sum([a[x][t] for t in a[x].keys()]), reverse=True)
+    #print(result_list)
+    for key in result_list:
+        val_list=sorted(a[key].keys(),key=lambda x : (a[key][x],-x), reverse=True)
+        #print(val_list)
+        
+        
+        for num,val in enumerate(val_list):
+            if num==2:
+                break
+            answer.append(val)
+
+            
+        
+        
+    return answer
```

Comment:
#include <string>
#include <vector>
#include <iostream>
#include <unordered_map>
#include <algorithm>

using namespace std;

vector<int> solution(vector<string> genres, vector<int> plays) {
    vector<int> answer;
    unordered_map <string ,unordered_map<int,int > > a;
    
    for (int i=0; i<genres.size();i++){
        a[genres[i]].insert(make_pair(i,plays[i]));
        
    }
    
    vector<string> keys;
    unordered_map <string ,int > genres_sum;
    for (auto pair : a){
        keys.push_back(pair.first);
        for(auto pair2 : a[pair.first]){
            genres_sum[pair.first]+=a[pair.first][pair2.first];
            
        }
    }
    
    sort(keys.begin(),keys.end(),  [&](string x, string y) -> bool {return genres_sum[x]>genres_sum[y]; }   );
        
    for (int i=0;i<keys.size();i++){
        vector<int> song_key;
         for (auto pair : a[keys[i]]){
        song_key.push_back(pair.first);
        }
        sort(song_key.begin(),song_key.end(),  [&](int x, int y) -> bool {return a[keys[i]][x]==a[keys[i]][y]? x<y: a[keys[i]][x]>a[keys[i]][y]; }   );
        
        
        for (int j=0;j<2 && j<song_key.size();j++){
            answer.push_back(song_key[j]);
            
        }
    }
    
    return answer;
}